src/Workspaces/Core/Portable/CodeFixesAndRefactorings/IFixAllContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using FixAllScope = Microsoft.CodeAnalysis.CodeFixes.FixAllScope;
+
+namespace Microsoft.CodeAnalysis.CodeFixesAndRefactorings
+{
+    internal interface IFixAllContext
+    {
+        IFixAllState State { get; }
+        IFixAllProvider? FixAllProvider { get; }
+        Solution Solution { get; }
+        Project Project { get; }
+        Document? Document { get; }
+        object Provider { get; }
+        FixAllScope Scope { get; }
+        string? CodeActionEquivalenceKey { get; }
+        CancellationToken CancellationToken { get; }
```

Comment:
note: teh CT being in FixAllContext was somewhat a mistake (and i remove it from 'FixAllState' intentionally so it would be captured by that).  Consider (not in this PR) removing this and making it so that we don't assume the context is responsible for storing the CT.

Reply:
Thanks, filed https://github.com/dotnet/roslyn/issues/60957